object/user.go

```diff
@@ -200,6 +202,17 @@ type ManagedAccount struct {
 	SigninUrl   string `xorm:"varchar(200)" json:"signinUrl"`
 }
 
+func (u *User) checkPasswordChangeRequiestAllowed() error {
+	if !u.isPasswordChangeRequiestAllowed() && u.PasswordChangeRequired {
+		return fmt.Errorf("PasswordChangeRequired not allowed for user '%s' due to be external one", u.Name)
+	}
+	return nil
+}
+
+func (u *User) isPasswordChangeRequiestAllowed() bool {
```

Comment:
What's this?

Reply:
since Casdoor does not support write back strategy for users from ldap or keycloak.
see Edit mode https://wjw465150.gitbooks.io/keycloak-documentation/content/server_admin/topics/user-federation/ldap.html.

So in this case no reason to ask user to change password - because after next sync pwd should be synced from 
Ldap/keycloak again.

If this validation looks needless I can remove it

Reply:
OK